### PR TITLE
Add escape-by-default to jelly files.

### PR DESCRIPTION
**Pull request by christ66**
@reviewbybees

### CODE REVIEW

File: src/main/resources/hudson/plugins/mercurial/MercurialChangeSetList/digest.jelly [hunk 1]
```diff
@@ -3,6 +3,7 @@
   when a build history link (or number) is followed
   e.g http://<hudson server>/job/<project>/<build number>/
 -->
+<?jelly escape-by-default='true'?>
```
Comment (jglick) [inline src/main/resources/hudson/plugins/mercurial/MercurialChangeSetList/digest.jelly]: Probably right but make sure `${cs.msgAnnotated}` is not _supposed_ to be HTML.

Comment (oleg-nenashev) [inline src/main/resources/hudson/plugins/mercurial/MercurialChangeSetList/digest.jelly]: Digged into the code for a bit. 
Jesse is right, the <code>msgAnnotated</code> returns HTML: https://github.com/jenkinsci/jenkins/blob/master/core/src/main/java/hudson/scm/ChangeLogSet.java#L248-260

**Comment by oleg-nenashev** [comment]
:bug:, see above
I would also expect more justification in PR description. Are there any real issues in Jenkins JIRA being caused by the missing escaping?

**Comment by oleg-nenashev** [comment]
@christ66 
ping, are there any plans?

**Comment by christ66** [comment]
@oleg-nenashev No issues being caused by `escape-by-default` not properly being set. Just ran an analysis tool and saw jelly file did not contain the option. Since there is no open issue in Jira for this we could close the pull request.

**Comment by oleg-nenashev** [comment]
@christ66 
I thinks it stills useful, so I would vote for finishing it for the sake of the better code quality and security. 

The right way would be to use <code>j:out</code> tag on the hazardous field. See https://wiki.jenkins-ci.org/display/JENKINS/Jelly+and+XSS+prevention

**Comment by oleg-nenashev** [comment]
I'm keeping the :bug: to keep the status. BTW, I think the fix efforts are minimal. If somebody want to release the new version, he should include this PR as well

**Comment by jglick** [comment]
Agreed, I think it just needs a final push. Probably suffices to revert fe14ac0ec1b7be0d1a46faf5a205713191a3ace6 and then

```diff
-${cs.msgAnnotated}
+<j:out value="${cs.msgAnnotated}"/>
```

plus sanity testing of course.

**Comment by tfennelly** [comment]
Fix the escaping of `${cs.msgAnnotated}` and it has a :bee: from me.

**Comment by oleg-nenashev** [comment]
:bee:

**Pull request merged**